components/slider/index.css

```diff
@@ -538,13 +539,19 @@ governing permissions and limitations under the License.
   overflow: hidden;
 }
 
-
+/* Applies to the filled-offset variant to keep track the same color for highcontrast mode */
 .spectrum-Slider-track {
   &::before {
-    background: var(--highcontrast-slider-track-color, var(--mod-slider-track-color, var(--spectrum-slider-track-color)));
+    background: var(--highcontrast-slider-track-color-static, var(--mod-slider-track-color, var(--spectrum-slider-track-color)));
   }
 }
 
+/* All variants other than filled-offset get a new track color for highcontrast mode */
+.spectrum-Slider-track:not(:has(~ .spectrum-Slider-fill)) {
```

Comment:
I agree that avoiding adding a class would be ideal, [last I was able to find in slack](https://adobedesign.slack.com/archives/G019JTYMT6H/p1678826333306739?thread_ts=1678821542.305869&cid=G019JTYMT6H) we were leaning away from using :has() was from March 2023 and the concerns were around Firefox. It does look like Firefox support has increased since then.

Reply:
The hesitancy was around some bugs with `:has` and shadow dom.  Did you test pulling this into the web component?

Reply:
@castastrophe great callout! I implemented this in SWC and the [PR is passing](https://github.com/adobe/spectrum-web-components/pull/3952). Are we good to move forward?